worlds/stardew_valley/stardew_rule/base.py

```diff
@@ -415,16 +435,13 @@ def get_difficulty(self):
 
     def __str__(self):
         if self.item not in self.other_rules:
-            return f"Has {self.item} -> {MISSING_ITEM}"
-        return f"Has {self.item}"
+            return f"Has {self.item} {self.group} -> {MISSING_ITEM}"
```

Comment:
I recommend some type of delimiter between item and group, for example putting the group in parenthesis. Since they can have spaces, otherwise it might not always be obvious where the item name and the group begins